com.unity.shadergraph/Tests/Editor/HeadlessPreview/PreviewTests.cs

```diff
@@ -485,5 +489,28 @@ public void MasterPreview_ValidShaderCode()
             //      7) Testing if a given property exists in the material property block?
             //      8) Testing if a property is at a given value in the MPB after setting the property
         }
+
+        static object[] interpreterTestCases = new object[]
+        {
+            ("Add1", new Color(1,0,0,1)), //Colors with Alpha 1 since target is opaque
+            ("Add2", new Color(0,1,0,1)),
+            ("Add3", new Color(1,1,0,1)),
+        };
+
+        [Test]
+        [TestCaseSource("interpreterTestCases")]
+        public void InterpreterTests((string nodeToCompile, Color expectedColor) input)
+        {
+            m_PreviewManager.SetActiveGraph(m_InterpreterTestsGraph);
+            m_PreviewManager.SetActiveRegistry(m_RegistryInstance);
+
+            var shaderString = m_PreviewManager.RequestNodePreviewShaderCode(m_InterpreterTestsGraph.GetNodeReader(input.nodeToCompile).GetName(), out var shaderMessages);
+            bool tmp = ShaderUtil.allowAsyncCompilation;
+            ShaderUtil.allowAsyncCompilation = false;
+            Shader shaderObject = ShaderUtil.CreateShaderAsset(shaderString, true);
+            ShaderUtil.allowAsyncCompilation = tmp;
+            var rt = DrawShaderToTexture(shaderObject);
+            Assert.AreEqual(input.expectedColor, rt.GetPixel(0, 0));
```

Comment:
These tests are failing on the error color: `RGBA(1.000, 0.000, 1.000, 1.000)`.

Reply:
```
ShaderGraphBlockOutput SYNTAX_Add1Main(ShaderGraphBlockInput In)
{
float1 SYNTAX_Add1_Out = float1(0, 0, 0, 0);
Add2_float1(SYNTAX_Add1_Out, **float1(1, 0, 0, 0)**, **float1(-1, 0, 0, 0)**);
ShaderGraphBlockOutput output;
output.BaseColor = SYNTAX_Add1_Out;
return output;
}
```

Looking deeper into it-- a float1 being called with 4 parameters is obviously a code-gen problem and would explain why all the tests above are failing also.

Reply:
Oh-- it's because I merged master into my local checkout of your branch and there are some changes to the way GraphType is setup.

You can probably sync with @brett-groshong on ensuring proper behavior there (I'm going to see if I can't make some adjustments to get them to run locally).

Reply:
Okay-- I found the bug. Brett and I updated the GraphType to use explicit enums for Length and Height. We forgot to update one spot where they are used. So this has nothing to do with your PR, but will need to be fixed on master.

`GraphType.GetShaderType` was not updated. After updating it, the tests pass!

Reply:
I applied the correction in a PR to sg2/staging-master here: https://github.com/Unity-Technologies/Graphics/pull/7068.

This will just make it so that when you merge master in your headlesspreview tests won't be borked. That they were breaking was my fault to begin with, I didn't run the headless preview tests when we merged the change.